src/App.tsx

```diff
@@ -15,36 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT = '',
+  ALPHABETICALLY = 'alphabetically',
+  LENGTH = 'length',
+}
+
+const getSortedGoods = (
+  goods: string[],
+  sortType: SortType,
+  reversed: boolean,
+): string[] => {
+  const sortedGoods = [...goods];
+
+  sortedGoods.sort((a, b) => {
+    switch (sortType) {
+      case SortType.ALPHABETICALLY:
+        return a.localeCompare(b);
+      case SortType.LENGTH:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortValue, setActiveButton] = useState<SortType>(SortType.DEFAULT);
+  const [reversed, setReversed] = useState<boolean>(false);
+
+  const sortGoods = (type: SortType) => {
+    setActiveButton(type);
+  };
+
+  const toggleReverse = () => {
+    setReversed(!reversed);
+  };
+
+  const resetGoods = () => {
+    setReversed(false);
+    setActiveButton(SortType.DEFAULT);
+  };
+
+  const sortedGoods = getSortedGoods(goodsFromServer, sortValue, reversed);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortValue !== SortType.ALPHABETICALLY,
+          })}
```

Comment:
[UX] - Don't forget to mark the active button to explain to the user what action is active. You've partially implemented this by changing the button class based on the `sortValue`, but ensure that the active state is clear and intuitive for the user.